File: Java/_056_Merge_Intervals/Solution.java
/**
 * Definition for an interval.
 * public class Interval {
 *     int start;
 *     int end;
 *     Interval() { start = 0; end = 0; }
 *     Interval(int s, int e) { start = s; end = e; }
 * }
 */
class Solution {
    public List<Interval> merge(List<Interval> intervals) {
        Collections.sort(intervals, new Comparator<Interval>() {
            @Override
            public int compare(Interval a, Interval b) {
                return a.start - b.start;
            }
        });
        List<Interval> result = new ArrayList<Interval>();
        if (intervals.size() == 0)
            return result;
        int start = intervals.get(0).start;
        int end = intervals.get(0).end;
        for (int i = 0; i < intervals.size() - 1; i++) {
            if (intervals.get(i + 1).start <= end)
                end = Math.max(end, intervals.get(i + 1).end);
            else {
                Interval temp = new Interval(start, end);
                result.add(temp);
                start = intervals.get(i + 1).start;
                end = intervals.get(i + 1).end;
            }
        }
        Interval temp = new Interval(start, end);
        result.add(temp);
        return result;
    }
}

File: Python/_056_Merge_Intervals/solution.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution:
    def merge(self, intervals):
        """
        :type intervals: List[Interval]
        :rtype: List[Interval]
        """
        if len(intervals) == 0:
            return []
        intervals = sorted(intervals,  key=lambda interval: interval.start)
        start = intervals[0].start
        end = intervals[0].end
        result = []
        for i in range(len(intervals) - 1):
            if intervals[i + 1].start <= end:
                end = max(end, intervals[i + 1].end)
            else:
                temp = Interval(start, end)
                result.append(temp)
                start = intervals[i + 1].start
                end = intervals[i + 1].end
        temp = Interval(start, end)
        result.append(temp)
        return result
